df_py/util/dftool_module.py

```diff
@@ -555,70 +568,78 @@ def do_calc():
             f"TOT_OCEAN was 0, so re-calc'd: TOT_OCEAN={TOT_OCEAN}"
             f", START_DATE={START_DATE}"
         )
-    elif START_DATE is not None:
-        print("TOT_OCEAN was nonzero, so re-calc'd: START_DATE=None")
-        START_DATE = None
-
-    # do we have the input files?
-    alloc_fname = csvs.allocationCsvFilename(CSV_DIR)  # need for loadStakes()
-    if not os.path.exists(alloc_fname):
-        print(f"\nNo file {alloc_fname} in '{CSV_DIR}'. Exiting.")
-        sys.exit(1)
-
-    vebals_fname = csvs.vebalsCsvFilename(CSV_DIR)  # need for loadStakes()
-    if not os.path.exists(vebals_fname):
-        print(f"\nNo file {vebals_fname} in '{CSV_DIR}'. Exiting.")
-        sys.exit(1)
-
-    if not csvs.nftvolsCsvFilenames(CSV_DIR):
-        print(f"\nNo 'nftvols*.csv' files in '{CSV_DIR}'. Exiting.")
-        sys.exit(1)
-
-    if not csvs.ownersCsvFilenames(CSV_DIR):
-        print(f"\nNo 'owners*.csv' files in '{CSV_DIR}'. Exiting.")
-        sys.exit(1)
-
-    if not csvs.symbolsCsvFilenames(CSV_DIR):
-        print(f"\nNo 'symbols*.csv' files in '{CSV_DIR}'. Exiting.")
-        sys.exit(1)
-
-    if not csvs.rateCsvFilenames(CSV_DIR):
-        print(f"\nNo 'rate*.csv' files in '{CSV_DIR}'. Exiting.")
-        sys.exit(1)
-
-    # shouldn't already have the output file
-    _exitIfFileExists(csvs.rewardsperlpCsvFilename(CSV_DIR, "OCEAN"))
-    _exitIfFileExists(csvs.rewardsinfoCsvFilename(CSV_DIR, "OCEAN"))
 
     # brownie setup
     networkutil.connect(5)
     ADDRESS_FILE = _getAddressEnvvarOrExit()
     recordDeployedContracts(ADDRESS_FILE)
 
-    # main work
-    S = allocations.loadStakes(CSV_DIR)
-    V = csvs.loadNftvolsCsvs(CSV_DIR)
-    C = csvs.loadOwnersCsvs(CSV_DIR)
-    SYM = csvs.loadSymbolsCsvs(CSV_DIR)
-    R = csvs.loadRateCsvs(CSV_DIR)
-    do_pubrewards = constants.DO_PUBREWARDS
-    do_rank = constants.DO_RANK
-
-    prev_week = 0
-    if START_DATE is None:
-        cur_week = calcrewards.getDfWeekNumber(datetime.datetime.now())
-        prev_week = cur_week - 1
-    else:
-        prev_week = calcrewards.getDfWeekNumber(START_DATE)
-    m = calcrewards.calcDcvMultiplier(prev_week)
-    print(f"Given prev_week=DF{prev_week}, then DCV_multiplier={m}")
+    if SUBSTREAM_NAME == "volume":
```

Comment:
I think each of the contents inside these ifs should go into their own modules as functions. i.e. volume/rewards.py, challenge/rewards.py. Dftool_module should only be a thin wrapper to the in-module work, in my opinion.

Reply:
Agree

Reply:
Closing this one as well, since there is another comment on this topic.